operational_analysis/methods/quality_check_automation.py

```diff
@@ -0,0 +1,331 @@
+
+    def indicesForCoord(self,f):
+
+        """
+        This function finds the nearest x/y indices for a given lat/lon. 
+        Rather than fetching the entire coordinates database, which is 500+ MB, this
+        uses the Proj4 library to find a nearby point and then converts to x/y indices.
+        This function relies on the Wind Toolkit HSDS API.
+        Args:
+        f (h5 file): file to be read in
+        Returns: x and y coordinates corresponding to a given lat/lon as a tuple
+        """
+
+        dset_coords = f['coordinates']
+        projstring = """+proj=lcc +lat_1=30 +lat_2=60 
+                    +lat_0=38.47240422490422 +lon_0=-96.0 
+                    +x_0=0 +y_0=0 +ellps=sphere 
+                    +units=m +no_defs """
+        projectLcc = Proj(projstring)
+        origin_ll = reversed(dset_coords[0][0])  # Grab origin directly from database
+        origin = projectLcc(*origin_ll)
+
+        lat, lon = self._lat_lon
+        coords = (lon, lat)
+        coords = projectLcc(*coords)
+        delta = np.subtract(coords, origin)
+        ij = [int(round(x/2000)) for x in delta]
+        return tuple(reversed(ij))
+
+    def ws_diurnal_prep(self, start_date = '2007-01-01', end_date = '2013-12-31'):
+
+        """
+        This method returns a Pandas Series corresponding to hourly average windspeeds
+
+        Args:
+        start_date(:obj:'String'): start date to diurnal analysis (optional)
+        end_date(:obj:'String'): end date to diurnal analysis (optional)
+
+
+        Returns:
+        ws_diurnal (Pandas Series): Series where each index corresponds to a different hour of the day and each value corresponds to the average windspeed
+        """
+
+        f = h5pyd.File("/nrel/wtk-us.h5", 'r')
+
+        # Setup date and time
+        dt = f["datetime"]
+        dt = pd.DataFrame({"datetime": dt[:]},index=range(0,dt.shape[0]))
+        dt['datetime'] = dt['datetime'].apply(dateutil.parser.parse)
+
+        project_idx = self.indicesForCoord(f)
+
+        print("y,x indices for project: \t\t {}".format(project_idx))
+        print("Coordinates of project: \t {}".format(self._lat_lon))
+        print("Coordinates of project: \t {}".format(f["coordinates"][project_idx[0]][project_idx[1]]))
+
+        # Get wind speed at 80m from the specified lat/lon
+        ws = f['windspeed_80m']
+        t_range = dt.loc[(dt.datetime >= start_date) & (dt.datetime < end_date)].index
+    
+        # Convert to dataframe
+        ws_tseries = ws[min(t_range):max(t_range)+1, project_idx[0], project_idx[1]]
+        ws_df=pd.DataFrame(index=dt.loc[t_range,'datetime'],data={'ws':ws_tseries})
+    
+        # Calculate diurnal profile of wind speed
+        ws_diurnal=ws_df.groupby(ws_df.index.hour).mean()
+
+        return ws_diurnal
+
+    def wtk_diurnal_plot(self):
+
+        """
+        This method plots the WTK diurnal plot alongisde the hourly power averages of the scada_df across all turbines
```

Comment:
Not necessarily scada_df, could be any power or wind speed data, no?

Reply:
Addressed